update substrate to polkadot-v0.9.40

koushiro opened the pull request.

koushiro commented:
@sorpaas PTAL

koushiro commented:
@sorpaas friendly ping

vedhavyas commented:
@koushiro I have bumped substrate rev further up and fixed some of the breaking changes. 

I have also removed `pallet-timestamp` over UnixTime. Would to like some feedback before I propose a change.
https://github.com/subspace/frontier/tree/subspace-v1

koushiro commented:
> @koushiro I have bumped substrate rev further up and fixed some of the breaking changes.
> 
> I have also removed `pallet-timestamp` over UnixTime. Would to like some feedback before I propose a change. https://github.com/subspace/frontier/tree/subspace-v1

The changes related to `pallet_timestamp` I think you can extract it separately and open a PR into upstream.

koushiro commented:
@sorpaas PTAL

koushiro commented:
@sorpaas PTAL

koushiro commented:
@sorpaas could you review and merge this PR? I wanna to update the substrate to polkadot-v0.9.42 ASAP.

koushiro commented:
CI broken on the master branch due to https://github.com/paritytech/frontier/pull/1031

koushiro commented:
> Hmm there are now more errors in tests..

it caused by #1030, I don't why no git conflicts occur 😕  I'm fixing it now
